include/bbp/sonata/reports.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+/**
+ * Provides a bridge between c-based programs in order to have access to the c++-based objects
+ */
+#ifndef __cplusplus
+#include <stddef.h>
+#endif
+#if defined(__cplusplus)
+#include <cstddef>
+#include <stdint.h>
+extern "C" {
+#endif
+
+
+/*! \brief Add node to existing or new report
+ * @param report_name name of the report to be created or added node to
+ * @param gid node/cell identifier
+ * @param tstart start time of recording for the report
+ * @param tend stop time of recording for the report
+ * @param dt frequency of recordings
+ * @param kind type of report (soma, compartment)
```

Comment:
Because of the way Neuron/CoreNeuron calls it. But as you mentioned in a previous comment I think this could be split in bindings/c while keeping a pure C++ interface here.